diff -rub libelf-0.8.11/lib/elf_repl.h libelf-0.8.11_new/lib/elf_repl.h
--- libelf-0.8.11/lib/elf_repl.h	2008-05-23 10:15:34.000000000 +0200
+++ libelf-0.8.11_new/lib/elf_repl.h	2009-07-07 14:02:31.000000000 +0200
@@ -973,6 +973,8 @@
     } c_un;
 } Elf32_Cap;
 
+#if __LIBELF64
+
 typedef struct {
     Elf64_Xword     	c_tag;
     union {
@@ -981,6 +983,8 @@
     } c_un;
 } Elf64_Cap;
 
+#endif
+
 #define CA_SUNW_NULL	0	/* c_un ignored */
 #define CA_SUNW_HW_1	1	/* c_un.c_val */
 #define CA_SUNW_SF_1	2	/* c_un.c_val */
